### PR TITLE
init: reorder trace points sequence

### LINKED ISSUE #1388: [FEATURE]trace point reorder in platform_init function
**Is your feature request related to a problem? Please describe.**
The trace point value now used in `platform_init` is out of order. After some refine of codec, there leave some blank trace point as well as miss trace point. The trace point is now misleading for new platform enable and QEMU debug.

**Describe the solution you'd like**
The trace point should be grouped and increase in order and did not miss any init function track.

### PR DESCRIPTION
**Pull request by tlauda**
Reorders trace points sequence to be incremental.
Also adds additional trace points for better debug.

Fixes #1388.

Signed-off-by: Tomasz Lauda <tomasz.lauda@linux.intel.com>

### CODE REVIEW

**Comment by dbaluta** [comment]
@tlauda  One patch per logical change please. Reorder and addition of new trace points should be separate patches.

File: src/arch/xtensa/smp/init.c [hunk 1]
```diff
@@ -94,13 +94,14 @@ int slave_core_init(struct sof *sof)
 	if (err < 0)
```
Comment (dbaluta) [inline src/arch/xtensa/smp/init.c]: Wat do you mean by " incremental trace points" can you enhance the message commit decription.

File: src/include/sof/trace.h [hunk 2]
```diff
@@ -35,30 +35,35 @@
  * when main trace is not available
  */
 #define TRACE_BOOT_START		0x1000
-#define TRACE_BOOT_ARCH		0x2000
+#define TRACE_BOOT_ARCH			0x2000
 #define TRACE_BOOT_SYS			0x3000
```
Comment (dbaluta) [inline src/include/sof/trace.h]: You are fixing indentation here, right? Please make it a separate patch.

File: src/include/sof/trace.h [hunk 3]
```diff
@@ -35,30 +35,35 @@
...
-#define TRACE_BOOT_PLATFORM_PMC		(TRACE_BOOT_PLATFORM + 0x130)
-#define TRACE_BOOT_PLATFORM_TIMER	(TRACE_BOOT_PLATFORM + 0x140)
-#define TRACE_BOOT_PLATFORM_CLOCK	(TRACE_BOOT_PLATFORM + 0x150)
-#define TRACE_BOOT_PLATFORM_SSP_FREQ	(TRACE_BOOT_PLATFORM + 0x160)
-#define TRACE_BOOT_PLATFORM_IPC		(TRACE_BOOT_PLATFORM + 0x170)
-#define TRACE_BOOT_PLATFORM_DMA		(TRACE_BOOT_PLATFORM + 0x180)
-#define TRACE_BOOT_PLATFORM_SSP		(TRACE_BOOT_PLATFORM + 0x190)
-#define TRACE_BOOT_PLATFORM_DMIC	(TRACE_BOOT_PLATFORM + 0x1a0)
-#define TRACE_BOOT_PLATFORM_IDC		(TRACE_BOOT_PLATFORM + 0x1b0)
+#define TRACE_BOOT_PLATFORM_IRQ		(TRACE_BOOT_PLATFORM + 0x110)
```
Comment (dbaluta) [inline src/include/sof/trace.h]: Not sure how this works, but I'm wondering if it is really OK to change the trace point numbers? Are there any tools depending on this exact numbers?

File: src/arch/xtensa/smp/init.c (hunk 1 above)
Comment (tlauda) [reply src/arch/xtensa/smp/init.c]: Done.

File: src/include/sof/trace.h (hunk 2 above)
Comment (tlauda) [reply src/include/sof/trace.h]: Done.

File: src/include/sof/trace.h (hunk 3 above)
Comment (tlauda) [reply src/include/sof/trace.h]: They are internal and not used by any tool, so it's OK.

**Comment by tlauda** [comment]
@dbaluta PR updated.

File: src/platform/intel/cavs/platform.c [hunk 4]
```diff
@@ -482,17 +480,17 @@ int platform_init(struct sof *sof)
...
 	trace_point(TRACE_BOOT_PLATFORM_IDC);
 	ret = idc_init();
 	if (ret < 0)
 		return ret;
 
+	/* init DAIs */
+	ret = dai_init();
+	if (ret < 0)
+		return ret;
+
```
Comment (dbaluta) [inline src/platform/intel/cavs/platform.c]: Did you actually move dai_init() later in the code? Or is git diff broken?

File: src/init/init.c [hunk 5]
```diff
@@ -44,6 +44,7 @@ int master_core_init(struct sof *sof)
 	init_heap(sof);
 
 #if CONFIG_TRACE
+	trace_point(TRACE_BOOT_SYS_TRACES);
 	trace_init(sof);
```
Comment (dbaluta) [inline src/init/init.c]: Does it make sense to call trace_point before trace_init?

File: src/platform/intel/cavs/platform.c (hunk 4 above)
Comment (tlauda) [reply src/platform/intel/cavs/platform.c]: Yes, I did.

File: src/init/init.c (hunk 5 above)
Comment (tlauda) [reply src/init/init.c]: It's different trace mechanism.

File: src/platform/intel/cavs/platform.c (hunk 4 above)
Comment (dbaluta) [reply src/platform/intel/cavs/platform.c]: @tlauda Ok, I understand. Then please add the change in a separate patch and explain why the movement was necessary.

Also, please make sure that the code compiles after each patch so that we won't break bisection in case of problems.

File: src/arch/xtensa/smp/init.c [hunk 6]
```diff
@@ -94,13 +94,13 @@ int slave_core_init(struct sof *sof)
...
```
Comment (dbaluta) [inline src/arch/xtensa/smp/init.c]: Also, please specify in this commit that we are actually also renumber the tracepoints in order to make place for new tracepoints.

File: src/init/init.c (hunk 5 above)
Comment (dbaluta) [reply src/init/init.c]: ok.

File: src/platform/intel/cavs/platform.c (hunk 4 above)
Comment (tlauda) [reply src/platform/intel/cavs/platform.c]: @dbaluta It's because we are changing ordering of the trace points as already stated in the commit. I understand that we should have as much granularity as possible, but I feel it's an overkill to split that kind of simple change into 4 separate commits.

Comment (dbaluta) [reply src/platform/intel/cavs/platform.c]: @tlauda ok, no strong preference on this change. Lets hope people will understand the reason behind this change from the commit message :).
